src/Umbraco.Infrastructure/Migrations/MigrationContext.cs

```diff
@@ -8,13 +10,15 @@ namespace Umbraco.Cms.Infrastructure.Migrations;
 /// </summary>
 internal class MigrationContext : IMigrationContext
 {
+    private readonly Action _onDoneAction;
     private readonly List<Type> _postMigrations = new();
 
     /// <summary>
     ///     Initializes a new instance of the <see cref="MigrationContext" /> class.
     /// </summary>
-    public MigrationContext(MigrationPlan plan, IUmbracoDatabase? database, ILogger<MigrationContext> logger)
+    public MigrationContext(MigrationPlan plan, IUmbracoDatabase? database, ILogger<MigrationContext> logger, Action onDoneAction)
     {
```

Comment:
Can `onDoneAction` be passed in `SetDone` instead, or without this method and call this action in caller only?